shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -35,14 +65,28 @@ public static <T extends Resource> Stream<T> resourcesInBundle(
                 .map(resource -> ((T) resource));
     }
 
-    public static <T extends Resource> Resource resourceInBundle(
-            Bundle bundle, Class<T> resourceType) {
+    public static <T extends Resource> T resourceInBundle(Bundle bundle, Class<T> resourceType) {
         return resourcesInBundle(bundle, resourceType).findFirst().orElse(null);
     }
 
+    // MSH - Message Header
+    public static MessageHeader getMSHMessageHeader(Bundle bundle) throws NoSuchElementException {
+        MessageHeader messageHeader = resourceInBundle(bundle, MessageHeader.class);
+        if (messageHeader == null) {
+            throw new NoSuchElementException("MessageHeader not found in the bundle");
+        }
+        return messageHeader;
+    }
+
+    public static MessageHeader createMSHMessageHeader(Bundle bundle) {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Hmm, good point. I think those creation methods will only be needed by tests

Reply:
Actually, moved back to this helper class now that is being used after refactoring to handle null when calling getMSHMessageHeader